### DISCUSSION TITLE
Generic method for byte classification?

### DISCUSSION THREAD

**Discussion by QuarticCat** [simdjson/simdjson#2033] [Q&A]
In a talk, @lemire mentioned that in simdjson he used a technique to classify bytes:

1. Construct 2 tables, each one has 16 entries, mapping a nibble (4-bit) to a byte.
2. Split a byte to upper nibble and lower nibble.
3. Lookup 2 nibbles in 2 tables respectively, and get 2 bytes.
4. Bitand / bitor these 2 bytes to get the final class of the original byte.

My question is, is there a generic algorithm to construct such tables and identify situations that are not applicable?

**Comment by lemire** [comment]
We call this vectorized classification. At this time, to my knowledge, it hasn't been formalized yet.

I will work on it.
